src/App.js

```diff
@@ -1,26 +1,94 @@
-import "./App.css";
-// STEP 4 - import the button and display components
-// Don't forget to import any extra css/scss files you build into the correct component
-
-// Logo has already been provided for you. Do the same for the remaining components
-import Logo from "./components/DisplayComponents/Logo";
+import React, { useState } from "react"
+import "./App.css"
+import { Numbers } from "./components/ButtonComponents/NumberButtons/Numbers"
+import Logo from "./components/DisplayComponents/Logo"
+import { Display } from "./components/DisplayComponents/Display"
+import { Operators } from "./components/ButtonComponents/OperatorButtons/Operators"
+import { Specials } from "./components/ButtonComponents/SpecialButtons/Specials"
 
 function App() {
-  // STEP 5 - After you get the components displaying using the provided data file, write your state hooks here.
-  // Once the state hooks are in place write some functions to hold data in state and update that data depending on what it needs to be doing
-  // Your functions should accept a parameter of the the item data being displayed to the DOM (ie - should recieve 5 if the user clicks on
-  // the "5" button, or the operator if they click one of those buttons) and then call your setter function to update state.
-  // Don't forget to pass the functions (and any additional data needed) to the components as props
-
-  return (
-    <div className="container">
-      <Logo />
-      <div className="App">
-        {/* STEP 4 - Render your components here and be sure to properly import/export all files */}
-      </div>
-    </div>
-  );
+	// STEP 5 - After you get the components displaying using the provided data file, write your state hooks here.
+	// Once the state hooks are in place write some functions to hold data in state and update that data depending on what it needs to be doing
+	// Your functions should accept a parameter of the the item data being displayed to the DOM (ie - should recieve 5 if the user clicks on
+	// the "5" button, or the operator if they click one of those buttons) and then call your setter function to update state.
+	// Don't forget to pass the functions (and any additional data needed) to the components as props
+	const [display, setDisplay] = useState("")
+	const [previousNumber, setPreviousNumberVal] = useState("")
+	const [operator, setOperator] = useState("")
+	const [clearDisplayOnNextOperation, setShouldClearDisplayOnNextOperation] = useState(false)
+
+	let numberOnClickHandler = function (value) {
+		if (clearDisplayOnNextOperation) {
+			setDisplay(value)
+			setShouldClearDisplayOnNextOperation(false)
+		} else {
+			setDisplay(display + value)
+		}
+	}
+
+	let operatorOnClickHandler = function (value) {
+		if (operator.length > 0 && operator !== "=") {
+			performOperation()
+			setOperator(value)
+		} else if (operator.length > 0 && operator === "=") {
+			performOperation()
+			setOperator("")
+		} else if (operator.length > 0) {
+			performOperation()
```

Comment:
I'm a little bit confused why you are performOperations() when the operator !== "=" and when it does === "="  and when the operator.length > 0.